Sources/Sentry/SentryFramesTracker.m

```diff
@@ -153,16 +153,15 @@ - (void)displayLinkCallback
 #        endif // defined(TEST) || defined(TESTCI)
     BOOL hasNoFrameRatesYet = self.frameRateTimestamps.count == 0;
     uint64_t previousFrameRate
-        = self.frameRateTimestamps.lastObject[@"frame_rate"].unsignedLongLongValue;
+        = self.frameRateTimestamps.lastObject[@"value"].unsignedLongLongValue;
     BOOL frameRateChanged = previousFrameRate != currentFrameRate;
     BOOL shouldRecordNewFrameRate
         = shouldRecordFrameRates && (hasNoFrameRatesYet || frameRateChanged);
     if (shouldRecordNewFrameRate) {
-        SENTRY_LOG_DEBUG(@"Recording new frame rate at %llu.", self.previousFrameSystemTimestamp);
-        [self.frameRateTimestamps addObject:@{
-            @"timestamp" : @(self.previousFrameSystemTimestamp),
-            @"frame_rate" : @(currentFrameRate),
-        }];
+        SENTRY_LOG_DEBUG(@"Recording new frame rate at %llu.", thisFrameSystemTimestamp);
+        [self recordTimestamp:thisFrameSystemTimestamp
+                        value:@(currentFrameRate)
+                        array:self.frameRateTimestamps];
```

Comment:
Previously slow/frozen frames were recorded with the same structure and dictionary keys (start and end timestamps), and refresh rates were different (timestamps and values). This PR unifies them so they all are recorded with a timestamp and a value: for slow/frozen frames, the value is the duration. For refresh rate changes, the value is the new refresh rate.

This also has a downstream effect of allowing us to consolidate `SentryProfiler.processFrameRenders` and `SentryProfiler.processFrameRates`, see later in the diff.